lib/ensure_content_type.js

```diff
@@ -72,8 +72,9 @@ function checkContentType(hyper, req, next, expectedContentType, responsePromise
                     expected: expectedContentType,
                     actual: res.headers['content-type']
                 });
-                // Limit response caching to a few seconds
-                res.headers['cache-control'] = 'max-age=10, s-maxage=10';
+                // Disable response caching, as we aren't setting a vary
+                // either.
+                res.headers['cache-control'] = 'max-age=0, s-maxage=0';
```

Comment:
They can get stuck only for a finite amount of time, and the frequency (and possibility) of that happening is rather low, so why disable caching altogether?